Mage/src/main/java/mage/abilities/keyword/AwakenAbility.java

```diff
@@ -120,19 +120,21 @@ public AwakenEffect copy() {
         @Override
         public boolean apply(Game game, Ability source) {
             UUID targetId = null;
-            for (Target target : source.getTargets()) {
-                if (target.getFilter().getMessage().equals(filterMessage)) {
-                    targetId = target.getFirstTarget();
+            if (source != null && source.getTargets() != null) {

```

Comment:
I wonder why the source can be null. There is probably anotehr problem causing this error we need to fix.
Or is there a possible scenario source can be null?